### PR TITLE
intro: Insert summary of partitioning oracle attacks

### PR DESCRIPTION
**Pull request by fscoto**
Continuation of #12.

---

CC @samuel-lucas6. This paragraph's been merged into the intro; I hope I understood the positioning correctly.

Additionally, we need to clear up the copyright situation. @samuel-lucas6 contributed the sequence "ability to craft a (ciphertext, authentication tag) pair that verifies under multiple keys significantly reduces the number of required interactions with the oracle in order to perform an exhaustive search, making it practical if the key space is small." which was taken wholesale into the merged suggestion. This is a sequence of words that may well meet the threshold of originality required for copyright coverage. This comes with rights, including the right to being recognized for your contribution by name. Therefore, I'd like to ask:

1. @samuel-lucas6, would you like to be listed as an author?
2. If no, would you like to be listed as an editor instead?
3. If either of the above were anwered with yes: Can we assume that you will make a pull request to make this addition after this PR is merged?
4. In any case to be on the safe side since it's non-obvious: Are you aware of and okay with your issue comments being contributions in the sense of IETF BCP 78, BCP 79 and the IETF Trust Legal Provisions (TLP) Relating to IETF Documents as noted CONTRIBUTING.md?

I completely understand that this is legalese nobody *really* wants to deal with, but I'd rather do it properly now than have to chase people months or years later.

### CODE REVIEW

File: draft-denis-aegis-aead.md [hunk 1]
```diff
@@ -70,7 +70,9 @@ This document describes the AEGIS-128L and AEGIS-256 authenticated encryption wi
...
+With some existing AEAD schemes, an attacker can generate a ciphertext that successfully decrypts under multiple different keys (partitioning oracle attack){{LGR21}}. This ability to craft a (ciphertext, authentication tag) pair that verifies under multiple keys significantly reduces the number of required interactions with the oracle in order to perform an exhaustive search, making it practical if the key space is small. One example for a small key space is password-based encryption: an attacker can guess a large number of passwords at a time by recursively submitting such a ciphertext to an oracle; this speeds up a password search by reducing it to a binary search.
+
+While this may be mitigated by means of inserting a padding block in the aforementioned algorithms, this workaround comes with additional processing cost and must itself be carefully constructed to resist leaking information via timing. As a key-committing AEAD scheme, the AEGIS cipher family is naturally resistant against padding oracle attacks.
```
Comment (jedisct1) [inline draft-denis-aegis-aead.md:75]: > against padding oracle attacks

against partitioning attacks :)

Comment (fscoto) [reply draft-denis-aegis-aead.md:75]: *How* do I keep doing this?!

**Comment by samuel-lucas6** [comment]
> I hope I understood the positioning correctly.

Looks good.

> Additionally, we need to clear up the copyright situation. @samuel-lucas6 contributed the sequence "ability to craft a (ciphertext, authentication tag) pair that verifies under multiple keys significantly reduces the number of required interactions with the oracle in order to perform an exhaustive search, making it practical if the key space is small." which was taken wholesale into the merged suggestion. This is a sequence of words that may well meet the threshold of originality required for copyright coverage. This comes with rights, including the right to being recognized for your contribution by name. Therefore, I'd like to ask:

That sentence was actually [Frank's](https://github.com/jedisct1/draft-aegis-aead/pull/12#issuecomment-974897147), but I wrote the 'an attacker can generate a ciphertext that successfully decrypts under multiple different keys' bit and suggested an example regarding the key space part.

I'm not sure that meets the threshold of originality, but if you think so, then I would happily be listed as an author or editor (whatever's more appropriate). I can do a PR and am fine with those contribution requirements.

I'm willing to help out more, but I doubt I'm needed and don't want to get in the way. You're both far more qualified than me.

**Comment by fscoto** [comment]
It is indeed not your sentence... and the part that's left may really be an issue with the threshold of originality. Here's my plan: I'm going to assume the authorship issue to be resolved with "no change to the list of authors". On the next discussion and/or PR you're involved in, we can revisit this. Is that okay with you?

> I'm willing to help out more, but I doubt I'm needed and don't want to get in the way. You're both far more qualified than me.

Qualification is, quite honestly, irrelevant to me personally. Good ideas come from anyone. If anything, having someone (who at least feels) less qualified means we can get input from someone who sees the draft from the position of what I'd informally call "a more typical implementer". I'd be happy to welcome you aboard.

**Comment by samuel-lucas6** [comment]
> Is that okay with you?

Yes, that's fine. I don't think it's enough for originality either.

> Qualification is, quite honestly, irrelevant to me personally. Good ideas come from anyone. If anything, having someone (who at least feels) less qualified means we can get input from someone who sees the draft from the position of what I'd informally call "a more typical implementer". I'd be happy to welcome you aboard.

True, although I wouldn't try to implement AEGIS myself just yet. Also, the RFC looks pretty sorted now.

**Pull request merged**